File: javaboxuegu/src/main/java/com/xushengling/javaboxuegu/utils/MD5Utils.java
package com.xushengling.javaboxuegu.utils;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * @ProjectName: BoXueGu
 * @Package: com.xushengling.javaboxuegu
 * @ClassName: MD5Utils
 * @Author: 徐圣领
 * @CreateDate: 2020/3/9 15:18
 */
public class MD5Utils {
    public static String md5(String text) {
        MessageDigest digest;
        try {
            digest = MessageDigest.getInstance("mu5");
            byte[] result = digest.digest(text.getBytes());
            StringBuilder sb = new StringBuilder();
            for (byte b : result) {
                String hex = Integer.toHexString(b & 0xff);
                if (hex.length()==1){
                    sb.append("0").append(hex);
                }else {
                    sb.append(hex);
                }
            }
            return sb.toString();
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
            return "";
        }
    }
}

File: app/src/main/java/com/xushengling/gitboxuegu/MD5Utils.kt
package com.xushengling.myapplication

import java.security.MessageDigest
import kotlin.experimental.and

/**
 *
 * @ProjectName:    BoXueGu
 * @Package:        com.xushengling.myapplication
 * @ClassName:      MD5Utils
 * @Author:         徐圣领
 * @CreateDate:     2020/3/9 14:58
 */
object MD5Utils {
    //MD5加密算法
    fun md5(text: String): String {
        val digest: MessageDigest?
        try {
            digest = MessageDigest.getInstance("md5")
            val result = digest.digest(text.toByteArray())
            val sb = StringBuilder()
            for (b in result) {
                val hex = Integer.toHexString((b and 0xff.toByte()).toInt())
                if (hex.length == 1) {
                    sb.append("0$hex")
                } else {
                    sb.append(hex)
                }
            }
            return sb.toString()
        } catch (e: EnumConstantNotPresentException) {
            e.printStackTrace()
            return ""
        }
    }
}